worlds/stardew_valley/strings/spells.py

```diff
@@ -1,22 +1,30 @@
+all_spells = []
+
+
+def spell(name: str) -> str:
+    all_spells.append(name)
+    return name
+
+
 class MagicSpell:
-    clear_debris = "Spell: Clear Debris"
-    till = "Spell: Till"
-    water = "Spell: Water"
-    blink = "Spell: Blink"
-    evac = "Spell: Evac"
-    haste = "Spell: Haste"
-    heal = "Spell: Heal"
-    buff = "Spell: Buff"
-    shockwave = "Spell: Shockwave"
-    fireball = "Spell: Fireball"
-    frostbite = "Spell: Frostbolt"
-    teleport = "Spell: Teleport"
-    lantern = "Spell: Lantern"
-    tendrils = "Spell: Tendrils"
-    photosynthesis = "Spell: Photosynthesis"
-    descend = "Spell: Descend"
-    meteor = "Spell: Meteor"
-    bloodmana = "Spell: Bloodmana"
-    lucksteal = "Spell: Lucksteal"
-    spirit = "Spell: Spirit"
-    rewind = "Spell: Rewind"
+    clear_debris = spell("Spell: Clear Debris")
```

Comment:
This works but it's a bit goofy imo,
String enum would be the best way (but I know there is some weirdness with those, especially between Python versions)

I would just make `MagicSpell` a dataclass and then have

```
from dataclasses import dataclass, fields

@dataclass
class MagicSpell:
    ...

all_spells = [field.name: for field in fields(MagicSpell)]
```

Reply:
I don't technically disagree with that, but there is a precedent (my fault, really) of that pattern in other stardew files, and these files are **intended** as constants, so it feels a bit weird to make it a data class and try to read its fields.

And also, a lot of that is changed in the 6.x.x branch anyway, so I personally wouldn't put much value in the exact formatting of this.
But since you set it as "request changes" and not "approved with comments", I'm guessing this means this is a blocker for you? If so, then yeah I guess we'll comply lol

Reply:
I mean both our approaches fill up the supposedly "constant" list programmatically :) I would argue this is in line with the purpose of dataclasses.fields

But I mean if it's like this in other places, then it's whatever. It's not incorrect or anything

Reply:
I also just realized my suggestion doesn't work. So, nevermind